pkg/packages/manager.go

```diff
@@ -77,6 +78,32 @@ func processInstalled(mc *ManagerContext) bool {
 		mc.Package.Status.State = api.StateUpdating
 		mc.RequeueAfter = retryShort
 		return true
+	} else {
+
+		var err error
+		newValues := make(map[string]interface{})
+		err = yaml.Unmarshal([]byte(mc.Package.Spec.Config), &newValues)
+		if err != nil {
+			mc.Log.Error(err, "unmarshaling current package configuration")
+			mc.Package.Status.Detail = err.Error()
+			mc.RequeueAfter = retryShort
+			return true
+		}
+
+		needs, err := mc.PackageDriver.IsConfigChanged(mc.Ctx, mc.Package.Name,
+			newValues)
+		if err != nil {
+			mc.Log.Error(err, "checking necessity of reconfiguration")
+			mc.Package.Status.Detail = err.Error()
+			mc.RequeueAfter = retryShort
+			return true
+		}
+		if needs {
+			mc.Log.Info("configuration change detected, upgrading")
+			mc.Package.Status.State = api.StateUpdating
+			mc.RequeueAfter = retryShort
+			return true
```

Comment:
tests maybe?

Reply:
I've added some in the latest push.

Reply:
Be nice to have a test a least for the "needs" case